src/coreclr/binder/customassemblybinder.cpp

```diff
@@ -242,21 +242,21 @@ void CustomAssemblyBinder::PrepareForLoadContextRelease(INT_PTR ptrManagedStrong
 CustomAssemblyBinder::CustomAssemblyBinder()
 {
     m_pDefaultBinder = NULL;
-    m_ptrManagedStrongAssemblyLoadContext = NULL;
+    m_ptrManagedStrongAssemblyLoadContext = 0;
```

Comment:
I worry that this kind of change reduces code readability? Are these new warnings related to a tooling update?

Reply:
With a more recent musl libc (the version in Alpine 3.19), NULL is defined as nullptr directly, making what was suppressable warnings with our current toolsets into unsupressable compiler errors. This PR includes fixes for all the locations that were now procuding errors for an x64 build on Alpine 3.19 (what the source-build pipeline tests).

In the future, I plan to follow up with re-enabling the warnings and fixing the remaining cases

Reply:
so is there no way to #ifdef NULL to 0 somehow for this platform? 

cc: @AaronRobinsonMSFT @janvorli

Reply:
This PR uses `(UINT_PTR)NULL` instead of 0 in a similar case, it seems like it would be better here too.

Reply:
Here it would be actually `(INT_PTR)NULL`